controllers/release/release_adapter.go

```diff
@@ -354,7 +354,8 @@ func (a *Adapter) registerGitOpsDeploymentStatus(binding *applicationapiv1alpha1
 	patch := client.MergeFrom(a.release.DeepCopy())
 
 	if condition.Status == metav1.ConditionTrue {
-		a.release.MarkDeployed(condition.Reason, condition.Message)
+		a.release.MarkDeployed(condition.Reason, condition.Message,
+			binding.CreationTimestamp, condition.LastTransitionTime)
```

Comment:
Why not to do like we do in the release? `MarkDeploying` function to set the start time and `MarkDeployed` to set the completion time.

Reply:
Sounds reasonable to me. We should track that information in the Release. We can restructure the status after this change so it's more readable though. Something like:
```
Conditions: []
Deployment:
    Binding: foo
    StartTime: ...
    CompletionTime: ...
    Target: bar
Release:
    PipelineRun: baz
    Strategy: qux
    StartTime: ...
    CompletionTime: ...
```